smarts/core/sumo_road_network.py

```diff
@@ -324,10 +324,15 @@ def oncoming_lanes_at_offset(self, offset: float) -> List[RoadMap.Lane]:
             if not nearby_lanes:
                 return result
             my_vect = self.vector_at_offset(offset)
-            my_norm = np.linalg.norm(my_vect)
+            my_norm = max(np.linalg.norm(my_vect), 1e-6)
```

Comment:
Do we use `1e-6` here to avoid divide by zero in line `lane_angle = np.dot(my_vect, lv) / (my_norm * np.linalg.norm(lv))` below?

If so, instead of doing `max(..., 1e-6)` here, can we just return empty `result=[]` if `np.linalg.norm(my_vect)=[0,0,0]` ?

Reply:
> Do we use `1e-6` here to avoid divide by zero in line `lane_angle = np.dot(my_vect, lv) / (my_norm * np.linalg.norm(lv))` below?

Yes, I think that was the intent there.

 
> If so, instead of doing `max(..., 1e-6)` here, can we just return empty `result=[]` if `np.linalg.norm(my_vect)=[0,0,0]` ?

Yes, I think this is an option -- at least when `my_norm == 0`, since when this happens, the lane at this offset doesn't really have a direction (probably because its length is (very near to) 0, then determining lanes that go in the "opposite" direction doesn't really make sense.  I would support either returning `[]` or `None` here as opposed to doing the `max()` in this case.

Between `[]` and `None`, I would probably prefer `None` in order to denote that this is an anomalous case, although that's a bit more painful for callers of this method, because then they'd have to check for it (which I doubt any do at this point).  Because of this, `[]` would also be ok, just because this event is hopefully going to be rare anyway.

Reply:
Ok i will go with returning `[ ]`.